RFCs/0051-exclude-operator.adoc

```diff
@@ -0,0 +1,1703 @@
+ | item_name | quantity |
+ +-----------+----------+
+ | sprocket  | 200      |
+ *-----------+----------*/
+----
+
+=== Snowflake
+* Reference: https://docs.snowflake.com/en/sql-reference/sql/select#selecting-all-columns-except-one-column
+* More info on Snowflake: https://dbdb.io/db/snowflake
+
+`SELECT * EXCLUDE` specifies the columns that should be excluded from the results. 
+
+
+[source,sql]
+----
+SELECT * EXCLUDE (department_id, employee_id) FROM employee_table;
+
++------------+------------+
+| LAST_NAME  | FIRST_NAME |
+|------------+------------|
+| Montgomery | Pat        |
+| Levine     | Terry      |
+| Comstock   | Dana       |
++------------+------------+
+----
+
+Also allows for selecting across multiple tables through repeated use of `EXCLUDE`:
+
+[source,sql]
+----
+SELECT table_a.* EXCLUDE column_in_table_a ,
+table_b.* EXCLUDE column_in_table_b
+...
+----
+
+=== DuckDB
+* Reference: https://duckdb.org/docs/sql/query_syntax/select
+* More info on DuckDB: https://dbdb.io/db/duckdb
+
+`SELECT * EXCLUDE` selects all the columns except the provided columns.
+
+[source,sql]
+----
+-- select all columns except the city column from the addresses table
+SELECT * EXCLUDE (city) FROM addresses;
+----
+
+=== Databricks
+* Reference: https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-qry-select.html#syntax
+
+Uses `SELECT * EXCEPT` to prune columns or fields from the referencable set of columns identified in the select_star clause. Worth mentioning that:
+
+Each name must reference a column included in the set of columns that you can reference or their fields. Otherwise, Databricks SQL raises a `UNRESOLVED_COLUMN` error. If names overlap or are not unique, Databricks SQL raises an `EXCEPT_OVERLAPPING_COLUMNS` error.
+
+
+Some examples:
+[source,sql]
+----
+-- select all referencable columns from all tables except t2.c4
+> SELECT * EXCEPT(c4) FROM VALUES(1, 2) AS t1(c1, c2), VALUES(3, 4) AS t2(c3, c4);
+  1   2   3
+
+-- select all referencable columns from a table, except a nested field.
+> SELECT * EXCEPT(c2.b) FROM VALUES(1, named_struct('a', 2, 'b', 3)) AS t(c1, c2);
+  1  { "a" : 2 }
+
+-- Removing all fields results in an empty struct
+> SELECT * EXCEPT(c2.b, c2.a) FROM VALUES(1, named_struct('a', 2, 'b', 3)) AS t(c1, c2);
+  1  { }
+
+-- Overlapping names result in an error
+> SELECT * EXCEPT(c2, c2.a) FROM VALUES(1, named_struct('a', 2, 'b', 3)) AS t(c1, c2);
+  Error: EXCEPT_OVERLAPPING_COLUMNS
+----
+
+=== MongoDB
+* Reference: https://www.mongodb.com/docs/manual/tutorial/project-fields-from-query-results/#return-all-but-the-excluded-fields
+* More info on MongoDB: https://dbdb.io/db/mongodb
+
+MongoDB supports excluding certain fields by setting the attribute's value to `0`. The following returns all fields other than the `status` and `instock` field for all of the matching documents.
+
+[source,shell]
+----
+db.inventory.find( { status: "A" }, { status: 0, instock: 0 } )
+----
+
+== Unresolved questions
```

Comment:
Wondering if partial schema case be out of scope only? i.e. having SQL's schema-full within the scope as well.